Calorie Tracker/Calorie Tracker/ViewControllers/CalorieTableViewController.swift

```diff
@@ -0,0 +1,173 @@
+//
+
+import UIKit
+import CoreData
+import SwiftChart
+
+class CalorieTableViewController: UITableViewController {
+    @IBOutlet private weak var chartView: Chart!
+
+    let calorieController = CalorieController()
+    
+    var caloriesArray = [Double]()
+
+    lazy var fetchedResultController: NSFetchedResultsController<Calorie> = {
+
+        let fetchRequest: NSFetchRequest<Calorie> = Calorie.fetchRequest()
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor(key: "timestamp", ascending: true)
+        ]
+        let moc = CoreDataStack.shared.mainContext
+        let frc = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: moc, sectionNameKeyPath: "calorie", cacheName: nil)
+        frc.delegate = self
+
+        do {
+            try frc.performFetch()
+        } catch {
+            print("Error fetching CoreDataStack: \(error)")
+        }
+
+        return frc
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        NotificationCenter.default.addObserver(self, selector: #selector(updateViews), name: .calorieEntered, object: nil)
+        updateViews()
+
+    }
+
+//    func dateString(for calorie: Calorie?) -> String? {
+//        calorie?.timestamp.map { dateFormatter.string(from: $0) }
+//    }
+    
+    @objc func updateViews() {
+        fetchedResultController.fetchedObjects?.forEach { calorie in
+            let calorie = calorie as Calorie
+            caloriesArray.append(Double(calorie.calorie))
+        }
+        
+        let series = ChartSeries(caloriesArray)
+        self.chartView.gridColor = .white
+        self.chartView.backgroundColor = .black
+        self.chartView.labelColor = .white
+        self.chartView.removeAllSeries()
+        self.chartView.add(series)
+        series.area = true
+        series.colors = (
+        above: ChartColors.cyanColor(),
+        below: ChartColors.redColor(),
+        //Makes it to where if you put in an abnormal calorie number for your intake the chart colors turn red
+        zeroLevel: 200
+        )
+    }
+    
+
+    @IBAction func addPressed(_ sender: Any) {
+        let alert = UIAlertController(title: "Add calorie intake", message: "Please enter number of calories", preferredStyle: .alert)
+        alert.addTextField { textField in
+            textField.placeholder = "Calorie Intake"
+        }
+        
+        alert.addAction(UIAlertAction(title: "OK", style: .default) { _ in
+            guard let calorieString = alert.textFields?.first?.text, !calorieString.isEmpty, let calories = Int16(calorieString) else { return }
+            self.calorieController.create(calorie: calories, date: Date())
+            self.caloriesArray.append(Double(calories))
+        })
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+        present(alert, animated: true)
+    }
+    override func numberOfSections(in tableView: UITableView) -> Int    {
```

Comment:
(A little reminder that removing this function will default to the tableView returning 1 section. )